openedx/core/lib/course_tabs.py

```diff
@@ -0,0 +1,47 @@
+"""
+Tabs for courseware.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+
+_ = lambda text: text
+
+
+# Stevedore extension point namespaces
+COURSE_TAB_NAMESPACE = 'openedx.course_tab'
+
+
+class CourseTabPluginManager(PluginManager):
+    """
+    Manager for all of the course tabs that have been made available.
+
+    All course tabs should implement `CourseTab`.
+    """
+    NAMESPACE = COURSE_TAB_NAMESPACE
+
+    @classmethod
+    def get_tab_types(cls):
+        """
+        Returns the list of available course tabs in their canonical order.
+        """
+        def compare_tabs(first_type, second_type):
+            """Compares two course tabs, for use in sorting."""
+            first_priority = first_type.priority
+            second_priority = second_type.priority
+            if not first_priority == second_priority:
+                if not first_priority:
```

Comment:
How are priorities represented?

Also, this catches when first_priority is falsy at all. Is that what you want? When first_priority is None and second_priority is 0, first_type will be returned as higher priority.